ui/src/core_components/CoreIFrame.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+	<div class="CoreIFrame" v-on:click="handleClick" ref="rootEl">
+		<iframe
+      @load="handleLoad" 
+			:src="fields.src.value"
+			draggable="false"
+		/>
+    <div class="mask" />
```

Comment:
This is because iframe intercepts all mouse events. According to docs first click should be to select and after that you are able to interact. This is why  i used mask. But you got me thinking and this probably won't work outside of the editor.

Reply:
Great point. We'll probably have to implement something that shows the mask only during edition time.

If you need to detect edit mode, you can use the injectable flag `isBeingEdited: Symbol() as InjectionKey<Ref<boolean>>`. For example, this is used in `CoreMessage.vue` to hide empty _Message_ components unless you're in edit mode.

There's likely some research to be done here; may involve `pointer-events` in CSS too, perhaps toggling it based on selection. You have the class `selected` injected at the root of the component when it's being selected.

So this might work, allowing the mask to capture the first click but being a passthrough for subsequent ones.

```css
.CoreIFrame.selected .mask {
  pointer-events: none;
}
```

Let's not rush and let's deliver a good experience that's consistent with the rest of the components.

Reply:
`pointer-events` is a good idea, I will add this.
Probably we don't need to relay on JS in this case, we could just use class `.buildMode` that is somewhere higher in the DOM tree, what do you think about that?
```
.CoreIFrame .mask {
	pointer-events: none;
}
.buildMode .CoreIFrame .mask {
	pointer-events: all;
}
```

Reply:
I think that wouldn't work because of scoped styles. Beyond that, I'd rather not rely on anything not injected directly by the `ComponentProxy`. I'm okay with adding class next to "selected", i.e. "beingEdited", if you prefer not to use CSS.

Also, the below would mean all of the clicks get captured by the mask, the first click and subsequent ones.

```css
.buildMode .CoreIFrame .mask {
	pointer-events: all;
}
```

Reply:
yup, I didn't copy all rules, sorry, here:
```css
.CoreIFrame .mask {
	pointer-events: none;
}
.buildMode .CoreIFrame .mask {
	pointer-events: all;
}
.CoreIFrame.selected .mask {
	pointer-events: none;
}
```
Rules are scoped but requirements for them are not, that means we can do something like above. and it doesn't require any additional code.
Don't know what is ComponentProxy yet 😛  `.buildMode` I found in `BuilderApp.vue`. 
Anyway if you want we can change it to `isBeingEdited` approach.

Reply:
You're right, let's keep it CSS as I like how clean it looks, but let's add a new `beingEdited` class to `ComponentProxy`.

Where it says 
```js
component: true,
childless: isChildless.value,
selected: isSelected.value,
disabled: isDisabled.value,
...fieldBasedCssClasses.value
```	

let's add `beingEdited: isBeingEdited.value`.

That'll inject the class when it's being edited. I'm sure it'll be useful down the line too.

So you'll be able to do `.CoreIFrame.beingEdited .mask` in the CSS.

Reason for this is that I want clear rules for component creation and the `.buildMode` wasn't really designed for this, so better not depend on it. My idea is for components to be as independent as possible and the `ComponentProxy` should serve as full interface between Streamsync and its different components (customs or core).

I think it's a good idea to get familiar with the `ComponentProxy`, it's like the secret sauce for Streamsync.

Reply:
Oh! Now I get it :D Good idea, will do!